### PR TITLE
Engine feature

### PR DESCRIPTION
**Pull request by marcb1309**
I added all three engines maia, zahak and rodentiv in the install process.

Files if needed download to DGTCentaurMods/tmp folder.
Engines and default uci files come in defaults folder and copied into engines folder as needed during install.
This will avoid personal changes to be deleted when dgtcentaurmods is removed and installed again.
tmp folder remains so that new install runs faster.

RodentIV can be configured on the page http://:5000/tuner - create character with self eplaining name and upload again. <- I know this could be done better - but it works.
RodentIV is a little special as it needs a reference to a personality file in rodentIV.uci (created on upload) and the personality.txt is then found in the personality folder. In addition so that the uci interface knows that the personality exists an entry is created in the personalities/basic.ini file.
currently the personality set "individual" is preselected. we could also add selectable diferent sets in the Web Gui.
The tuner page is not very fancy and I hide it on purpose.
Maybe we should also add a default character. Otherwise it might be weird if you select the Rodent engine and there is nothing to select?!

game/menu.py is changed so that is only shows the files but not folders.

I added functionality to the build process to speed up Stockfish addition if already there. If there it will ask if we want to rebuild or not.

Last but not least - The build.sh must be adopted when the pull request is accepted so that is load the repository from it´s original location.

Greets.
Marc

### CODE REVIEW

File: build/build.sh [hunk 1]
```diff
@@ -2,34 +2,48 @@
 set -e
 # Script to produce deb pacjage
 
-##### VARIABLESi
+##### VARIABLES
 BASE=`pwd`
 REPO_NAME="DGTCentaur"
-REPO_URL="https://github.com/EdNekebno/DGTCentaur"
+#REPO_URL="https://github.com/EdNekebno/DGTCentaur"
+REPO_URL="-b engine_feature https://github.com/marcb1309/DGTCentaur"
```
Comment (wormstein) [inline build/build.sh:9]: This needs to be restored.

File: build/control [hunk 2]
```diff
@@ -1,5 +1,5 @@
 Package: dgtcentaurmods
-Version: 0
+Version: 0.0.1
```
Comment (wormstein) [inline build/control:2]: This gets changed by build.sh anyway depending on repo tag. Is there a reason for 0.0.1? Wobt affect sed anyway, so I think it looks better anyway :)

**Comment by wormstein** [comment]
> I added all three engines maia, zahak and rodentiv in the install process.
> 
> Files if needed download to DGTCentaurMods/tmp folder. Engines and default uci files come in defaults folder and copied into engines folder as needed during install. This will avoid personal changes to be deleted when dgtcentaurmods is removed and installed again. tmp folder remains so that new install runs faster.
> 
I don't think this will preserve configuration. On remove of the package, everything gets wiped out now. We don't have an update process yet. I think for "marketing" purpose we can just integrate those engines by default but keep the additional books and weights download as you did, on setup. Put a safety check after git clone if it fails, remove the engine and throw an alert to the user.

> RodentIV can be configured on the page http://:5000/tuner - create character with self eplaining name and upload again. <- I know this could be done better - but it works. RodentIV is a little special as it needs a reference to a personality file in rodentIV.uci (created on upload) and the personality.txt is then found in the personality folder. In addition so that the uci interface knows that the personality exists an entry is created in the personalities/basic.ini file. currently the personality set "individual" is preselected. we could also add selectable diferent sets in the Web Gui. The tuner page is not very fancy and I hide it on purpose. Maybe we should also add a default character. Otherwise it might be weird if you select the Rodent engine and there is nothing to select?!
> 
I would rename this to "Rodent config" or something, not just tuner. Same as above, just add the engine and keep additional downloads separated.

These changes will also streamline the setup a bit as we don't bug the user with to many questions.

> game/menu.py is changed so that is only shows the files but not folders.
> 
> I added functionality to the build process to speed up Stockfish addition if already there. If there it will ask if we want to rebuild or not.
> 
SF should not be there anyway as every build should start after ```make clean```. The fact the stage folder remains there is a miss, I just wanted to be able to peek inside for debugging. I could add an oprion to build command to just remove the old .deb and re-use the stage folder as it is already there. Then your change will fit as SF might be in it!

> Last but not least - The build.sh must be adopted when the pull request is accepted so that is load the repository from it´s original location.
> 
> Greets. Marc
Any questions, let me know. 
All the best, Adrian.

File: build/build.sh (hunk 1 above)
Comment (wormstein) [reply build/build.sh:9]: Ignore this. Will updated on merge...

**Comment by marcb1309** [comment]
The apt remove or dpkg -r only remove the files added through the deb install process itself  which means the files that are copied over from default folder into engines during postinst script will remain after uninstall. Except we would declare a purge routine to delete all folders after uninstall. 
I can clearly remove all questions during the installation process which wasn't really the best course of action anyway. So I keep all four engines as the default in the installation process. In the end, everyone is free to delete these engines and folders using the web GUI after installation.

I will add some safety check for the git clone process needed to download additional files for RodentVI and maia weigth files.

**Comment by wormstein** [comment]
Makes sense now, I just didn't get the logic of that will save the config. But what about on reinstall? Would the dpkg skip any existing files/folders? I am also exploring new grounds, that's why I ask some things...

In this case, let's rename "defaults" to "setup" and have an "engines" folder inside. At the end of the setup we just wipe the folder. It could contain in the future other usefull stuff...

Let me know if all these changes takes to much of your time and I could spread the load by doing the changes after merging. I just want to have your input before doing changes on your code...

The rest looks good.

-A.

**Comment by marcb1309** [comment]
Yes the postinst script only copies from default (setup) folder into the engine folder when files do not yet exist. That ensures the uci files and personalities/basic.ini file will survive an uninstall and install.

**Comment by marcb1309** [comment]
RodentIV can be configured on the page http://:5000/rodentivtuner

Engines itself are now in the engines folder whereas uci and basic.ini are in /setup/engines folder and copied into engines folder on setup so that they survive an uninstall.

**Pull request merged**